### ISSUE TITLE
Pivotted Inverse solve and potentially inverse is incorrect when lhs matrix is complex valued

### ISSUE THREAD

**Issue by romeric** [romeric/Fastor#110]
Consider the following

~~~c++
using T = std::complex<double>;
constexpr size_t M = 3;
Tensor<size_t,M> p;
Tensor<T,M,M> L, U, P;

Tensor<T,M,M> A;
for (int i = 0; i < M; ++i)
{
    for (int j = 0; j < M; ++j) {
        A(i, j) = T(i,j);
        if (i==j) A(i, j) *= 1000;
    }
}
~~~

Inverse 
~~~c++
Tensor<T,M,M> invA = inverse(A);
~~~
gives
~~~
[        (-333.666583249917,333.666583249917),     (0.333166499833166,0.000166833500166857),    (0.333416750083417,-8.34167500834149e-05)]
[  (-0.000166833500166857,-0.333166499833166), (0.000333667000333667,-0.000333667000333667), (-0.000166833500166834,0.000166833500166834)]
[   (8.34167500834149e-05,-0.333416750083417), (-0.000166833500166834,0.000166833500166834), (8.34167500834168e-05,-8.34167500834168e-05)]
~~~

which is the correct answer, while pivotted inverse 
~~~c++
Tensor<T,M,M> invA = inverse<InvCompType::SimpleInvPiv>(A);
~~~
gives

~~~
[(8.34167500834168e-05,-8.34167500834168e-05), (-0.000166833500166834,0.000166833500166834),    (8.34167500834149e-05,-0.333416750083417)]
[(-0.000166833500166834,0.000166833500166834), (0.000333667000333667,-0.000333667000333667),   (-0.000166833500166857,-0.333166499833166)]
[   (0.333416750083417,-8.34167500834149e-05),     (0.333166499833166,0.000166833500166857),         (-333.666583249917,333.666583249917)]
~~~

which is incorrect.

**Comment by romeric** [comment]
This is not a complex valued number issue, even on double precision

~~~c++
using T = double;
constexpr size_t M = 3;
Tensor<size_t,M> p;
Tensor<T,M,M> L, U, P;

Tensor<T,M,M> A;
for (int i = 0; i < M; ++i)
{
    for (int j = 0; j < M; ++j) {
        A(i, j) = T(i+j);
        if (i==j) A(i, j) *= 1000;
    }
}
~~~

pivotted inverse is incorrect

**Comment by romeric** [comment]
Post multiplying the pivot matrix with inverse solved the issue for now 3a16475. Need to be reworked for performance.
